### PR TITLE
Development info and Dockerfile

### PR DESCRIPTION
**Pull request by JPrevost**
- provides general overview of dev environment
- provides basic Dockerfile
- provides defaults via .flaskenv

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM python:3.7-slim
```
Comment (matt-bernhardt) [inline Dockerfile]: I haven't kept track of what our use of various Python versions is, but know that 3.8 was released recently. Do we have a strategy for what version we use?

Comment (JPrevost) [reply Dockerfile]: Well... the app is on 3.7 hence the docker image is on 3.7. However, is there a good reason the app is not on 3.8? Absolutely not :)

Comment (matt-bernhardt) [reply Dockerfile]: Fair enough :-)

Comment (JPrevost) [reply Dockerfile]: <cough>
https://github.com/MITLibraries/renorect/issues/4
</cough>

**Comment by matt-bernhardt** [comment]
This is probably some failure on my part, but I'm unable to reproduce the instructions. The issue comes in the final step - there is a tip shown about installing `python-dotenv` in order to read values from `.flaskenv`. The docker run command then ends, saying it could not locate a Flask application.

The Pipenv graph doesn't seem to include this library - is that supposed to be at the system level, or should it be part of what pipenv installs?

**Comment by JPrevost** [comment]
@matt-bernhardt no, you are correct. I must have made a change inadvertently. I'll do a fixup and let you know when it's ready for review.

**Comment by gravesm** [comment]
I'm wondering if the docker build is even necessary here? It's being deployed on heroku so we don't need it for deployment. You could just use a makefile with a single target to start the server. Unless there's more that's going to be added to this app, skipping the container seems way easier.

**Comment by JPrevost** [comment]
@gravesm I have been leaning towards having a way for developers with little experience with a language being able to get an app up and running quickly locally without having to necessarily understand the best way to get specific versions of languages installed initially. It's my current feeling, and I'm happy to take it to Slack to dig in deeper on this, that a Dockerfile that runs the application should be part of our engineering best practices. That said, I can absolutely see how an app like this may break that (currently unwritten) rule even if we wrote it.

I had previously (and still kind of do) support a vagrant that can (in theory) run all of our ruby and python apps with little fuss. However, it's a bit clunky and I'd prefer to move to app specific Dockerfiles.

**Pull request merged**